torch/csrc/dynamo/guards.cpp

```diff
@@ -59,11 +56,12 @@ class TensorCheck {
     if (ndim != dim_) {
       return false;
     }
-    if (!dynamic_shapes_) {
-      const auto& sizes = v.sizes();
-      const auto& strides = v.strides();
-      for (auto i : c10::irange(ndim)) {
-        if (sizes_[i] != sizes[i] || strides_[i] != strides[i]) {
+    const auto& sizes = v.sizes();
+    const auto& strides = v.strides();
+    for (auto i : c10::irange(ndim)) {
+      auto known_size = sizes_[i];
+      if (known_size.has_value()) {
+        if (known_size.value() != sizes[i] || strides_[i] != strides[i]) {
```

Comment:
I think you need some more tests... won't this always reject dynamic shapes where the dynamic size is on the right hand side of the tensor? All of the strides will vary in that case too and static only stride test would always fail.